README.md

```diff
@@ -22,23 +22,121 @@ At this point you can either run the repo using Docker or directly with Ruby.
 
 Using Docker, run the following command:
 
-* `docker-compose up`
+- `docker-compose up`
 
 This starts up a web server at <http://localhost:4000>, builds the static site and allows you to view changes live.
 
 _This has been tested with Docker version 2.3.0.3. See [Docker installation instructions](https://docs.docker.com/get-docker/)._
 
 _Note: On Windows you will need to re-run the `docker-compose up` command to see changes, unless you are using [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10)._
 
+## Publishing to JSBin
+
+This configuration is not required for basic documentation creation and modification, so most
+editors can simply skip to [Forking and running locally](#forking-and-running-locally).
+
+If you will be creating or modifying code within this repository that is to be uploaded automatically
+to [JSBin](https://jsbin.ably.io/), for our "Try it now code editor", then you will need to create yourself a JSBin config file:
+
+### Ably staff members
+
+To obtain the API key you need to run this command in the terminal, you must ensure you have installed the hero-cli tools (Instructions on [installing the Heroku toolbelt](https://devcenter.heroku.com/articles/heroku-cli) )
+
+```bash
+$ heroku pg:psql -a ably-jsbin -c "select name, api_key from ownership"
+```
+
+When you execute this command a browser window will open and you will need to login using your **Ably email address**, only Ably staff members will be granted permission to access this resource.
+
+If you are a staff member and you are receiving this message
+
+```bash
+$ heroku pg:psql -a ably-jsbin -c "select name, api_key from ownership"
+ ▸    You do not have permission to view addon resources on ably-jsbin. You need to
+ ▸    have the deploy or operate permission on this app.
+```
+
+then ask your admin to check you have permission on Heroku, then re-authenticate with.
+
+```bash
+$ heroku login
+```
+
+If everything goes smoothly you will be given the API key.
+
+```
+ name |      api_key
+------+-------------------
+ ably | 12345678901234567
+(1 row)
+```
+
 ### Running with Ruby
 
 Using Ruby, run the following command:
 
-* `bundle install` 
+```bash
+$ cp config/jsbin_config.example.yaml config/jsbin_config.yaml
+```
+
+### Compile the JSBin code snippets
+
+Once you have finished creating your code snippet you will need to add the new assets to the existing ones. This is essentially a table of hashed documents and corresponding file names, [./jsbins.yaml](./jsbins.yaml) if you are interested in the entire list.
+
+```bash
+$ bundle exec nanoc compile
```

Comment:
Can we also give the docker version of this? I just re-ran docker-compile up to do this.

Reply:
what is the latest version?